WordPress/src/main/java/org/wordpress/android/ui/prefs/AccountSettingsFragment.java

```diff
@@ -328,6 +337,48 @@ public void onAccountChanged(OnAccountChanged event) {
         }
     }
 
+    /**
+     * If the username can be changed then the control can be clicked to open to the
+     * Username Changer screen.
+     */
+    private void checkIfUsernameCanBeChanged() {
+        AccountModel account = mAccountStore.getAccount();
+        mUsernamePreference.setEnabled(account.getUsernameCanBeChanged());
+        mUsernamePreference.setOnPreferenceClickListener(preference -> {
+            showUsernameChangerFragment(account);
+            return false;
+        });
+    }
+
+    private void showUsernameChangerFragment(AccountModel account) {
+        final Bundle bundle =
+                SettingsUsernameChangerFragment.newBundle(account.getDisplayName(), account.getUserName());
+
+        new FullScreenDialogFragment.Builder(getActivity())
+                .setTitle(R.string.username_changer_title)
+                .setAction(R.string.username_changer_action)
+                .setOnConfirmListener(this)
+                .setHideActivityBar(true)
+                .setOnDismissListener(null)
+                .setContent(SettingsUsernameChangerFragment.class, bundle)
+                .build()
+                .show(((AppCompatActivity) getActivity()).getSupportFragmentManager(), FullScreenDialogFragment.TAG);
+    }
+
+    @Override public void onConfirm(@Nullable Bundle result) {
+        if (result != null) {
+            String username = result.getString(BaseUsernameChangerFullScreenDialogFragment.RESULT_USERNAME);
+
+            if (username != null) {
+                ToastUtils.showToast(getActivity(),
+                        String.format(getString(R.string.settings_username_changer_toast_content), username),
+                        ToastUtils.Duration.LONG);
+
+                refreshAccountDetails();
```

Comment:
I was thinking to set the `Username` preference with the username here since I have the value, but I have realized that all other changes in the settings simply call `refreshAccountDetails` so I reused that approach. Open to thoughts on this :)